src/GoogleFit/FitOnFhir.GoogleFit.Tests/GoogleFitImportServiceTests.cs

```diff
@@ -180,26 +180,33 @@ public async Task GivenEventDataBatchTryAddReturnsFalse_WhenMedTechDatasetAdded_
         }
 
         [Fact]
-        public void GivenDatasetRequestThrowsException_WhenProcessDatasetRequestsIsCalled_ExceptionIsThrown()
+        public async Task GivenDatasetRequestThrowsException_WhenProcessDatasetRequestsIsCalled_AggregateExceptionIsThrown()
         {
-            string exceptionMessage = "DatasetRequest exception";
-            var datasetRequestException = new Exception(exceptionMessage);
-
             _googleFitUser.TryGetLastSyncTime(_dataStreamId, out Arg.Any<long>())
                 .Returns(x =>
                 {
                     x[1] = _oneDayBack;
                     return true;
                 });
 
+            string exceptionMessage = "DatasetRequest exception";
+            var datasetRequestException = new Exception(exceptionMessage);
             _googleFitClient.DatasetRequest(
                 Arg.Is<string>(access => access == _tokensResponse.AccessToken),
                 Arg.Any<DataSource>(),
                 Arg.Any<string>(),
                 Arg.Any<int>(),
                 Arg.Is<CancellationToken>(token => token == _cancellationToken)).Throws(datasetRequestException);
 
-            Assert.ThrowsAsync<Exception>(async () => await _googleFitImportService.ProcessDatasetRequests(_googleFitUser, _dataSources, _tokensResponse, _cancellationToken));
+            try
+            {
+                await _googleFitImportService.ProcessDatasetRequests(_googleFitUser, _dataSources, _tokensResponse, _cancellationToken);
+            }
+            catch (AggregateException ex)
+            {
+                Assert.NotNull(ex.InnerException);
+                Assert.Equal(datasetRequestException.Message, ex.InnerException.Message);
```

Comment:
You are going to want to move these checks outside of the catch. If this fails to throw an exception or if the exception is not AggregateException this test will pass. You can either use a finally or just put it outside of the catch.